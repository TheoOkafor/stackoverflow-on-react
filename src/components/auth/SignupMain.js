import React, { Component } from 'react';
import SignupForm from './SignupForm';

class SignupMain extends Component {
  constructor(props) {
    super(props);

    this.state = {
      data: [],
    };
  }

  render() {
    return (
      <div className="main row-container">
        <div className="sd-margin-12 d-margin-6 margin-top-20 site-desc">
          <h1 className="heading center">
            Join Stack Overflow<small>-lite</small>
          </h1>
          <p>
            This is a platform where you can ask
            <span>questions</span> and provide <span>answers</span>.
          </p>
        </div>

        <SignupForm />
        <div className="clear-fix" />

      </div>
    );
  }
}

export default SignupMain;
