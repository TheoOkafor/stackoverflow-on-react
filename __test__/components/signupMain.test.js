import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import SignupMain from '../../src/components/auth/SignupMain';
import store from '../../src/store';

test('Test that the Signup Main component displays', () => {
  const component = renderer.create(
    <Provider store={store}>
      <SignupMain />
    </Provider>
  );
  expect(component).toMatchSnapshot();
});
