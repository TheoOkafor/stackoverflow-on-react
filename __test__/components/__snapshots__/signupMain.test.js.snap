// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that the Signup Main component displays 1`] = `
<div
  className="main row-container"
>
  <div
    className="sd-margin-12 d-margin-6 margin-top-20 site-desc"
  >
    <h1
      className="heading center"
    >
      Join Stack Overflow
      <small>
        -lite
      </small>
    </h1>
    <p>
      This is a platform where you can ask
      <span>
        questions
      </span>
       and provide 
      <span>
        answers
      </span>
      .
    </p>
  </div>
  <div
    className="sd-margin-12 d-margin-6"
  >
    <div
      className="sign-card"
    >
      <form
        className="signup-form"
        onSubmit={[Function]}
      >
        <p>
          <small>
            Already Signed up?
          </small>
          <a
            className="inherit"
            href="/signin"
          >
            Sign in
          </a>
        </p>
        <p
          className="signup-error "
        >
          
        </p>
        <div
          className="form-item"
        >
          <label>
            Username
          </label>
          <input
            name="username"
            onChange={[Function]}
            placeholder="username"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-item"
        >
          <label>
            Email
          </label>
          <input
            data-testid="email"
            name="email"
            onChange={[Function]}
            placeholder="Your current email address"
            required={true}
            type="email"
          />
        </div>
        <div
          className="form-item"
        >
          <label>
            Create Password
          </label>
          <input
            data-testid="password"
            id="password"
            name="password"
            onInput={[Function]}
            placeholder="Create Password"
            required={true}
            type="password"
          />
          <h5
            className=""
            id="pass-strength"
          >
            
          </h5>
        </div>
        <div
          className="form-item"
        >
          <label>
            Repeat Password
          </label>
          <input
            data-testid="password2"
            id="password2"
            name="confirmpassword"
            onChange={[Function]}
            placeholder="Repeat password"
            required={true}
            type="password"
          />
          <h5
            id="match-display"
          >
            
          </h5>
        </div>
        <button
          className="btn primary"
          data-testid="signupBtn"
          id="sign-up"
        >
          Sign Up
        </button>
      </form>
    </div>
  </div>
  <div
    className="clear-fix"
  />
</div>
`;
